doc/source/dev/debugging_galaxy_slurm.md

```diff
@@ -0,0 +1,81 @@
+
+# Debugging Galaxy: Slurm Compute Cluster
+
+This document explains how to debug a Galaxy instance that uses a Slurm compute cluster (not the
+default LocalRunner). If you have a MacOS machine, you need to create a Linux VM using Oracle's 
+VirtualBoxVM and run the instructions in the VM. For reference, I created a 64-bit Linux VM with 
+4096 MB of RAM and 32 GB of disk space (all other parameters were left as default) and managed to 
+successfully run the instructions. If using a Linux VM, you may need to install Git, Python, VSCode, 
+and Docker (if using Irods storage). Finally, These instructions are intended for developers who 
+want to debug their Galaxy instance and are not meant for deployment purposes (We have nice Ansible 
+playbooks for that).
+
+## Debugging Galaxy in VS Code 
+
+1. Install libslurm-dev
+    * `sudo apt install libslurm-dev`
+    * On my VM, the above command puts slurm/slurm.h in /usr/include and libslurm.a in /usr/lib/x86_64-linux-gnu
+    * You need these header and library files to compile slurm-drmaa in the next step
+
+2. Install slurm-drmaa
+    * Download slurm-drmaa tar file from https://github.com/natefoo/slurm-drmaa/releases/download/1.1.2/slurm-drmaa-1.1.2.tar.gz
+    * Extract the downloaded tar file into home directory: `tar -xzvf slurm-drmaa-1.1.2.tar.gz -C ~`       
+    * To compile and install slurm-drmaa
+        * `cd ~/slurm-drmaa-1.1.2`
+        * `./configure --with-slurm-inc=/usr/include --with-slurm-lib=/usr/lib/x86_64-linux-gpu`
+        * `sudo make install`
+        * On my VM, the above command puts libdrmaa.so in /usr/local/lib. You need the path to this file in the next step
+
+3. Replace the contents of ~/galaxy/config/job_conf.xml with the following code snippet (create the file if it does not already exist)
+    ```
+    <?xml version="1.0"?>
+    <job_conf>
+        <plugins workers="3">
+            <plugin id="slurm" type="runner" load="galaxy.jobs.runners.slurm:SlurmJobRunner">
+                <param id="drmaa_library_path">/usr/local/lib/libdrmaa.so</param>
+            </plugin>
+        </plugins>
+        <destinations default="slurm">
+            <destination id="slurm" runner="slurm"/>
+        </destinations>
+        <resources default="default">
```

Comment:
This block shouldn't be here if you're not explicitly using it.

Reply:
I'll remove the resources block then. Thx.